isinstance and catch exception

randdane opened the pull request:
Added a check of instance for clarity and added catch for error if attachment path not valid.

randdane committed:
isinstance and catch exception

Added a check of instance for clarity and added catch for error if attachment path not valid.

randdane commented:
I hope to look more into starttls() soon.

ekozlowski commented on simpleMail.py:
```diff
@@ -67,11 +67,15 @@ def send_email(frm, to, subject, message, attachments=None, maxsize=100000, html
+        try:
+            att.set_payload(open(attachment, 'rb').read())
+        except OSError as err:
+            print('OS Error: {}'.format(err))
```
The handling of an OSError exception here will not allow any OSError exceptions to percolate upward, resulting in the email message being sent whether or not the attachment was successfully attached.  IMO, the exception should be raised, and the calling code should handle the exception by not asking the file(s) to be attached, or perhaps resending the email with the proper path to the attachment.

randdane replied:
What about sending the exception into the email as text? It seems like this might be nice if you want to ensure you get the email. Instead of an invalid attachment you just get some text.

```
except OSError as err:
    att = MIMEText('OS Error: {}'.format(err))
    msg.attach(att)
```

randdane replied:
I just noticed that I could put `msg.attach(att)` in a separate `finally:` block.

ekozlowski commented on simpleMail.py:
```diff
@@ -49,7 +49,7 @@ def send_email(frm, to, subject, message, attachments=None, maxsize=100000, html
     # if we're passed a string on 'to', don't join with commas.  If we're passed a list,
     # we'll need to join the adddresses with commas.
-    if not hasattr(m_to, 'upper'):
+    if isinstance(m_to, list):
```
Nice.  This is cleaner.  I've always used hasattr(<obj>, 'upper'), because other instances of items could be iterable, besides lists. :+1:

randdane committed:
Attachment error sent as text.

If the path to an attachment is invalid, an OSError is raised and sent as text in the email instead of the attachment continuing.

randdane committed:
Added finally block to error handling.
